### DISCUSSION TITLE
Flash-programming attached microcontrollers

### DISCUSSION THREAD

**Discussion by rtek1000** [arendst/Tasmota#14254] [Q&A] [answered]
Hi,

Would it be possible to add some functions like Esp-Link?

(transparent bridge / flash-programming)

> The esp-link firmware connects a micro-controller to the internet using an ESP8266 Wifi module. It implements a number of features:
> 
> transparent bridge between Wifi and serial, useful for debugging or inputting into a uC
> flash-programming attached Arduino/AVR microcontrollers and LPC800-series and other ARM microcontrollers via Wifi
> built-in stk500v1 programmer for AVR uC's: program using HTTP upload of hex file
> outbound REST HTTP requests from the attached micro-controller to the internet
> MQTT client pub/sub from the attached micro-controller to the internet
> serve custom web pages containing data that is dynamically pulled from the attached uC and that contain buttons and fields that are transmitted to the attached uC (feature not fully ready yet)

https://github.com/jeelabs/esp-link

**Comment by Jason2866** [comment, marked as answer]
Take a look at Tasmota Client https://tasmota.github.io/docs/TasmotaClient/
